src/styles/main.scss

```diff
@@ -1,3 +1,65 @@
+* {
+  font-family: Arial, sans-serif;
+}
+
+$height-of-block: 100px;
+$width-of-block: 100px;
+$amount-of-days-in-row: 7;
+$num-days: 31;
+$start-days: 0 "mon", 1 "tue", 2 "wed", 3 "thu", 4 "fri", 5 "sat", 6 "sun";
+$month-length: 28, 29, 30, 31;
+
 body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+@for $i from 1 through $num-days {
+  .calendar {
+    display: flex;
+    gap: 1px;
+    flex-wrap: wrap;
+    max-width: $height-of-block * $amount-of-days-in-row + $amount-of-days-in-row;
+
+    &__day {
+      box-sizing: border-box;
+      height: $height-of-block;
+      width: $width-of-block;
+      background-color: #eee;
+      border: 1px black solid;
+      text-align: center;
+      line-height: $height-of-block;
+      font-size: 30px;
+      transition: all 0.5s ease;
+
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+
+      &:hover {
+        background-color: pink;
+        transform: translate(0, -20px);
+        cursor: pointer;
+      }
+    }
+
+  }
+}
+
+
+@each $index, $day in $start-days {
+  .calendar--start-day--#{$day} :first-child {
+    margin-left: 100px * $index + $index;
+  }
+}
+
+@each $index in $month-length {
+  @for $i from $index through $num-days {
+    .calendar--month-length--#{$index} :nth-child(#{$i + 1}) {
+      opacity: 0;
```

Comment:
it is better to use display: none here. I still can hover over hidden days and have cursor pointer